dbc_extract3/dbc/generator.py

```diff
@@ -2856,6 +2863,11 @@ def generate(self, ids = None):
             fields += category.field('dmg_class')
             f, hfd = category.get_hotfix_info(('dmg_class', 47))
 
+            # 48
+            targeting = spell.get_link('targeting')
+            fields += targeting.field('max_affected_targets')
+            f, hfd = targeting.get_hotfix_info(('max_affected_targets', 48))
+
```

Comment:
You should be able to do something like:
```
targeting = spell.children('SpellTargetRestrictions')
if len(targeting):
    fields += targeting[0].field('max_affected_targets')
    f, hfd = targeting[0].get_hotfix_info(('max_affected_targets', 48))
else:
    fields += self.db('SpellTargetRestrictions').default().field('max_affected_targets')
```

This is kinda clunky though, so I'm also slightly wondering if that children() interface should change to return a default entry if there are no actual children available for the parent. Note also that children() will return a list of entries, because technically, a single parent record can have multiple child records in a db2 file.

Reply:
That's much nicer, yeah. I decided to go a bit further and introduced a `.child()` accessor. Not sure... but for stuff that we either know there is only 1 thing (or we simply don't care) it seems it's much easier.

Reply:
Yeah that's fine, although it does hide problems if blizzard makes an unconventional spell data change (adds another child record and the ordering is such that the old one is no longer the first entry).

Reply:
I don't think we would be able to handle such a change for stuff we expect a 1-to-1 mapping even with an explicit `children()[0]` access. It would also hide the problem from us.

What I think we can do for data we expect a 1-to-1 link is to slap a warning inside `child()` in case the number of linked records is more than one. It would actually better protect us from changes in spell data.

Reply:
No, of course not, better way would be to analyze the data before generation and warn about multiple entries.